src/torchscale/architecture/config.py

```diff
@@ -86,6 +86,7 @@ def postprocessing(self):
 
 
 class DecoderConfig(PretrainedConfig):
+    model_type = 'custom_transformer'
```

Comment:
I don't completely understand why we need this now, when we didn't need it before.

Reply:
I don't know why the error just started popping up now and I don't know enough about HuggingFace's automodels/autoconfigs libraries to say either. It may have been an update to HF transformers library that brought this on, though we're not using a new version, so idk.

Anyway, I was getting the below error running it, and it was clear that the decoder config was missing this element (the RetNet had it, and wasn't throing the same error). So I added this line and the code works. It might work if we call it a transformer, in which case we could just call it model type from the config, but I think that would err out b/c the LongNet uses the same registered class as the normal transformer. So hardcoding it into the class avoids that issue and naming it `custom_transformer` seems like a fair middle ground for transformer and LongNet.

```ValueError… The config you are passing has a 'model_type' attribute that is not consistent with the model type you passed (config has and you passed custom_transformer. Fix one of those so they match!```